src/Resolution.ts

```diff
@@ -265,14 +286,26 @@ export default class Resolution {
    *  - BTC
    *  - ETH
    * @throws [[ResolutionError]] if address is not found
+   * @depricated since v1.6.2 use Resolution.addr instead
    */
   async addressOrThrow(
     domain: string,
     currencyTicker: string,
   ): Promise<string> {
+    console.warn(
+      'Resolution#addressOrThrow is depricated since 1.6.2, use Resolution#addr instead',
+    );
     domain = this.prepareDomain(domain);
     const method = this.getNamingMethodOrThrow(domain);
-    return await method.address(domain, currencyTicker);
+    try {
+      return await method.addr(domain, currencyTicker);
+    } catch(error) {
+      // re-throw an error for back compatability. New method.addr throws deprecated UnspecifiedCurrency code since v1.6.2
+      if (error instanceof ResolutionError && error.code === ResolutionErrorCode.RecordNotFound) {
+        throw new ResolutionError(ResolutionErrorCode.UnspecifiedCurrency, {domain, currencyTicker, deprecated: true});
+      }
+      throw error;
+    }
```

Comment:
You've removed all the tests for `#addressOrThrow` and changed its implementation at the same time. Usually, people keep tests for deprecated methods until they are 100% removed. I would be ok if you delete them but wouldn't change the implementation. In your case: did you check that `#addressOrThrow` tests pass before you changed them to `#addr`?

Reply:
Yes, I am 100% it works, and nothing breaks. To get this pr a little more trustworthiness I added some of the old tests with addressOrThrow in Ens.test.ts. I guess we are getting close to releasing v2.0.0? we already have 1 configuration input, 4 functions, and 1 error code deprecated.